cpp/src/io/orc/stripe_init.cu

```diff
@@ -419,21 +409,13 @@ extern "C" __global__ void __launch_bounds__(128, 8)
   uint32_t chunk_id         = blockIdx.y * num_columns + blockIdx.x;
   int t                     = threadIdx.x;
 
-  if (t < sizeof(ColumnDesc) / sizeof(uint32_t)) {
-    // NOTE: Assumes that sizeof(ColumnDesc) <= 128x4
-    ((volatile uint32_t *)&s->chunk)[t] = ((const uint32_t *)&chunks[chunk_id])[t];
-  }
-  __syncthreads();
-  if (strm_info) {
-    int strm_len = s->chunk.strm_len[t >> 6];
-    int strm_id  = s->chunk.strm_id[t >> 6];
-    int t6       = t & 0x3f;
-    if (strm_len > 0 && t6 < sizeof(CompressedStreamInfo) / sizeof(uint32_t)) {
-      ((volatile uint32_t *)&s->strm_info[t >> 6])[t6] =
-        ((const uint32_t *)&strm_info[strm_id])[t6];
-    }
-  }
   if (t == 0) {
+    s->chunk = chunks[chunk_id];
+    if (strm_info) {
+      s->strm_info[0] = strm_info[s->chunk.strm_id[0]];
+      s->strm_info[1] = strm_info[s->chunk.strm_id[1]];
+    }
```

Comment:
I'm having trouble seeing where the 
```
s->strm_info[1] = strm_info[s->chunk.strm_id[1]];
```
line comes from in the original code. Was this added to fix a bug?

Reply:
This kernel is of size {128, 1, 1} and `(>>6)` means `/64`,  so earlier two sets of values were being copied. So, I just replaced it with direct update.

Reply:
`sizeof(CompressedStreamInfo)` looks to be 64 bytes and divided by `sizeof(uint32_t)=4` is 16. 
So doesn't `t6` range from `0-15`?

Reply:
But, we are concerned about `s->strm_info[t >> 6]`, which would be only two sets of values in 128 threads